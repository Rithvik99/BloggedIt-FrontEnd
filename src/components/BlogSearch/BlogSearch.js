import { AppBar, Button, FormHelperText, Grid, TextField, Typography } from '@material-ui/core';
import ChipInput from 'material-ui-chip-input';
import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { useHistory, useLocation } from 'react-router-dom';
import { getBlogBySearch } from '../../actions/blogs';
import Blogs from '../Blogs/Blogs';
import useStyles from './styles';

function useQuery() {
  return new URLSearchParams(useLocation().search);
}

const BlogSearch = () => {
  const [search, setSearch] = useState('');
  const [tags, setTags] = useState([]);
  const classes = useStyles();
  const dispatch = useDispatch();
  

  const [currentId, setCurrentId] = useState(null);
  const query = useQuery();
  const history = useHistory();
  const searchQuery = query.get('searchQuery');
  const [bp, setbp] = useState(false);

  const searchBlogs = () => {
    setbp(true)
    if (search.trim() || tags.length > 0) {
      dispatch(getBlogBySearch({ search, tags: tags.join(',') }));
      history.push(`/blogs/search?searchQuery=${search || 'none'}&tags=${tags.join(',')}`);
    } else {
      history.push('/');
    }
  };


  const handleAdd = (tag) => setTags([...tags, tag]);
  const handleDelete = (tagToDelete) => setTags(tags.filter((tag) => tag !== tagToDelete));

  return (
    <div className= {classes.container}>
      <div className= {classes.searchcontainer}>
        <AppBar className={classes.appBarSearch} position="static" color="inherit">
        <Typography variant="h5"> Dive into a world of informative and engaging blogs.</Typography>
          <TextField
            className={classes.searchtext}
            name="search"
            variant="outlined"
            label="Search Blogs"
            fullWidth
            value={search}
            id="search-input"
            // onKeyPress={(e) => { handleKeyPress(e) }}
            onChange={(e) => {
              setSearch(e.target.value);
            }}
          />
          <ChipInput
            className={classes.tag}
            value={tags}
            onAdd={handleAdd}
            onDelete={handleDelete}
            label="Search Tags"
            variant="outlined"
          />
          <FormHelperText className={classes.helptext}>Press Enter after entering your tag</FormHelperText>
          <Button onClick={searchBlogs} className={classes.searchButton} variant="contained" color="primary">Search</Button>
        </AppBar>
      </div>
      <div className= {classes.Blogscontainer}>
      {(bp &&
        <Grid container justifyContent="center">
          <Grid item xs={12} sm={6} md={9}>
            <Blogs setCurrentId={setCurrentId}/>
          </Grid>
        </Grid>
      )}
      </div>
    </div>
  );
};

export default BlogSearch;
