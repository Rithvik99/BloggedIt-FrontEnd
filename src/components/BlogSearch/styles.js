import { makeStyles } from '@material-ui/core/styles';

export default makeStyles((theme) => ({
  appBarSearch: {
    borderRadius: 20,
    marginBottom: theme.spacing(4),
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    padding: theme.spacing(4),
    
  },
  
  container: {
    display: 'flex',
    flexDirection: 'column',
    // alignItems: 'center',
  },
  tag: {
    margin: theme.spacing(1),
    marginBottom: theme.spacing(1),
    width: '90%',
    maxWidth: 400, 
  },
  searchtext: {
    margin: theme.spacing(2),
    marginTop: theme.spacing(4),
    width: '90%',
    maxWidth: 400, 
  },
  helptext: {
    marginBottom: theme.spacing(2),
  },
  blogsContainer: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: theme.spacing(4),
  },
}));