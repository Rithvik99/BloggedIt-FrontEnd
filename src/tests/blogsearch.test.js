/** @jest-environment jsdom */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import BlogSearch from '../components/BlogSearch/BlogSearch';

// Mock dependencies
jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn(),
    useDispatch: jest.fn(),
  }));
  
jest.mock('react-router-dom', () => ({
...jest.requireActual('react-router-dom'),
useHistory: jest.fn(),
}));

// Mock location
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useLocation: () => ({
        search: '?searchQuery=none&tags=none',
    }),
}));

// Mock search action
jest.mock('../actions/blogs', () => ({
    ...jest.requireActual('../actions/blogs'),
    getBlogBySearch: jest.fn(),
}));

// Test if the BlogSearch component renders correctly
test('BlogSearch renders correctly', () => {
    const { container: view, getByRole } = render(<BlogSearch/>);
    expect(view).toMatchSnapshot();
});