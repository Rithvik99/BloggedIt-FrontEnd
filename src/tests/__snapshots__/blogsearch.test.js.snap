// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogSearch renders correctly 1`] = `
<div>
  <div
    class="makeStyles-container-2"
  >
    <div>
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorInherit makeStyles-appBarSearch-1 MuiPaper-elevation4"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
        >
           Dive into a world of informative and engaging blogs.
        </h5>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-searchtext-4 MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
            for="search-input"
            id="search-input-label"
          >
            Search Blogs
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="search-input"
              name="search"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-9"
              >
                <span>
                  Search Blogs
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root makeStyles-tag-3 WAMuiChipInput-root-11"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root WAMuiChipInput-outlined-15 WAMuiChipInput-label-19 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
          >
            Search Tags
          </label>
          <div
            class="WAMuiChipInput-outlined-15 WAMuiChipInput-chipContainer-14 WAMuiChipInput-labeled-18"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root WAMuiChipInput-inputRoot-12 WAMuiChipInput-outlined-15 MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input WAMuiChipInput-input-13 WAMuiChipInput-outlined-15 MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-8"
                  style="width: 0.01px;"
                >
                  <span>
                    â€‹
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <p
          class="MuiFormHelperText-root makeStyles-helptext-5"
        >
          Press Enter after entering your tag
        </p>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Search
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </header>
    </div>
    <div />
  </div>
</div>
`;
